packages/server/src/provider/provider.ts

```diff
@@ -58,4 +66,12 @@ export interface Provider extends CommonProvider<ServerProviderStatus> {
     context: EvaluationContext,
     logger: Logger,
   ): Promise<ResolutionDetails<T>>;
+
+  /**
+   * Track a user action or application state, usually representing a business objective or outcome.
+   * @param trackingEventName
+   * @param context
+   * @param trackingEventDetails
+   */
+  track?(trackingEventName: string, context?: EvaluationContext, trackingEventDetails?: TrackingEventDetails): void;
```

Comment:
This is the same for server and client, I think this could live in the `CommonProvider` like `initialize` and `onClose`.

Reply:
Oh you are right I think. Will do.